src/Controls/src/Core/Border/Border.cs

```diff
@@ -64,7 +64,7 @@ void NotifyStrokeShapeChanges()
 
 			if (strokeShape is VisualElement visualElement)
 			{
-				AddLogicalChild(visualElement);
+				SetInheritedBindingContext(visualElement, BindingContext);
```

Comment:
@PureWeen maybe you remember if there is something this change would break? This makes `Border.StrokeShape` and `Border.Stroke` the same now.

Reply:
This should be fine

Prior to converting this to `AddLogicalChild` we were already setting the parent
https://github.com/dotnet/maui/blame/cd7a0e021d99be3ef8c7d83437fb66f5080cd5af/src/Controls/src/Core/Border/Border.cs#L67-L68

Which is why I converted it over to `AddLogicalChild` during that big push

Stephanes changes with these changes should make this all work alright

Reply:
do we need to copy the concept with parent resource propagation that's part of stroke?

https://github.com/dotnet/maui/blob/2a7abf457f6091edd113251b92d177e26da0f652/src/Controls/src/Core/Border/Border.cs#L113-L114

Reply:
For `Stroke`, I guess you could have a `GradientBrush` that uses a `{StaticResource}` for a color inside. That seems like it would be required.

For `StrokeShape`, I guess you can do the exact thing for something like a `double` value on some shape, so I'll make the properties the same.